src/core_landice/mode_forward/mpas_li_thermal.F

```diff
@@ -1079,6 +1082,14 @@ subroutine li_thermal_solver(domain, err)
                           basalEnthalpy,             &
                           basalTemperature(iCell))
 
+                    
+                     basalConductiveFlux(iCell) = -iceConductivity/thickness(iCell) * &
+                        (basalTemperature(iCell)-temperature(nVertLevels,iCell))/(1.0_RKIND - layerCenterSigma(nVertLevels))
+                    ! TZ: The basal conductive flux should be calculated using temperature, not
+                    ! enthalpy. If we use enthalpy, basalConductiveFlux will always equals to basal heat flux, causing problem
+                    ! At ice surface, it is unlikely there will be temperate ice. So enthalpy can just be used
```

Comment:
Yes, it is confusing. I think that's what I meant. If at the surface the ice is temperate, i.e., containing certain water fraction, we need to modify the surface boundary condition a bit. Currently it's H_s = T_s * rho_i * cp_i, assuming the ice is cold or having no water inside. We need to change it a bit to H_s = T_s * rho_i * cp_i + w*L (w is the water content and L is the latent heat of fusion of water). However, we can't know accurately the water content, and for most simulations we conduct for ice sheets, the annual mean temperature is < 0. So I would just stick with the old code and don't make any modifications at this point.

Reply:
Ok, I think that makes sense to me to leave the specified temp. b.c., at least until some future time when we are able to investigate it more carefully.  Can you make an additional note here explaining that a bit further?  Specifically, the most important thing is that we choose to leave water content out of the boundary condition because we aren't confident in assigning a value to it.

Reply:
Sure. I'll put more comments there.

Reply:
I've added some explanations in the code